data-prepper-core/src/main/java/org/opensearch/dataprepper/core/peerforwarder/PeerForwardingProcessorDecorator.java

```diff
@@ -101,7 +101,11 @@ public Collection<Record<Event>> execute(final Collection<Record<Event>> records
         final Collection<Record<Event>> recordsSkipped = new ArrayList<>();
         for (Record<Event> record: records) {
             if (((RequiresPeerForwarding)innerProcessor).isApplicableEventForPeerForwarding(record.getData())) {
-                recordsToProcess.add(record);
+                if (isPeerForwardingDisabled()) {
```

Comment:
Nit: This condition could just be a part of the above if statement

```
if (!isPeerForwardingDisabled && ((RequiresPeerForwarding)innerProcessor).isApplicableEventForPeerForwarding(record.getData()))
```

Reply:
I thought about it but the next condition has `isForLocalProcessingOnly(record.getData())){` which will fail. This is more clean